hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/multinetwork/MultiNetworkIsolationSuite.java

```diff
@@ -0,0 +1,129 @@
+    @org.junit.jupiter.api.BeforeEach
+    void initDefaults() {
+        setConfigDefaults();
+    }
+
+    static {
+        // Use a distinct shard/realm for NET_C to prove isolation is insensitive to shard/realm overlap
+        System.setProperty("hapi.spec.multinet.NET_C.shard", "21");
+        System.setProperty("hapi.spec.multinet.NET_C.realm", "22");
+    }
+
+    @MultiNetworkHapiTest(
+            networks = {
+                @MultiNetworkHapiTest.Network(name = "NET_A", size = 1),
+                @MultiNetworkHapiTest.Network(name = "NET_B", size = 1),
+                @MultiNetworkHapiTest.Network(name = "NET_C", size = 1)
+            })
+    @DisplayName("Three network isolation")
+    Stream<DynamicTest> threeNetworkIsolation(
+            final SubProcessNetwork netA, final SubProcessNetwork netB, final SubProcessNetwork netC) {
+        final var acctA = new AtomicReference<String>();
+        final var acctB = new AtomicReference<String>();
+        final var acctC = new AtomicReference<String>();
+        final long balA = ONE_HBAR;
+        final long balB = 2 * ONE_HBAR;
+        final long balC = 3 * ONE_HBAR;
+
+        final var builder = multiNetworkHapiTest(netA, netB, netC)
+                .onNetwork(
+                        "NET_A",
+                        TxnVerbs.cryptoCreate("acctA")
+                                .balance(balA)
+                                .exposingCreatedIdTo(id -> acctA.set(asAccountString(id))),
+                        QueryVerbs.getAccountBalance(acctA::get).hasTinyBars(balA))
+                .onNetwork(
+                        "NET_B",
+                        TxnVerbs.cryptoCreate("acctB")
+                                .balance(balB)
+                                .exposingCreatedIdTo(id -> acctB.set(asAccountString(id))),
+                        QueryVerbs.getAccountBalance(acctB::get).hasTinyBars(balB))
+                .onNetwork(
+                        "NET_C",
+                        TxnVerbs.cryptoCreate("acctC")
+                                .balance(balC)
+                                .exposingCreatedIdTo(id -> acctC.set(asAccountString(id))),
+                        QueryVerbs.getAccountBalance(acctC::get).hasTinyBars(balC))
+                // Even with the same shard/realm and colliding account numbers, each network
+                // should return its own balance for the "foreign" account id.
+                .onNetwork(
+                        "NET_A",
+                        // NET_A and NET_B share shard/realm; querying acctB resolves to NET_A's local account id
+                        QueryVerbs.getAccountBalance(acctB::get).hasTinyBars(balA),
+                        // NET_C uses different shard/realm, so its id should be invalid here
+                        QueryVerbs.getAccountBalance(acctC::get)
+                                .hasAnswerOnlyPrecheck(ResponseCodeEnum.INVALID_ACCOUNT_ID))
```

Comment:
Maybe it's the late hour, but I'm not following exactly how these overlapping shard/realms work. Will ask about it tomorrow